zerver/tests/test_message_dict.py

```diff
@@ -644,3 +644,33 @@ def test_sew_messages_and_reaction(self) -> None:
             self.assertEqual(data["reactions"][0]["emoji_name"], "simple_smile")
             self.assertTrue(data["id"])
             self.assertTrue(data["content"])
+
+    def test_reaction_order(self) -> None:
+        sender = self.example_user("othello")
+        receiver = self.example_user("hamlet")
+        recipient = Recipient.objects.get(type_id=receiver.id, type=Recipient.PERSONAL)
+        sending_client = make_client(name="test suite")
+
+        message = Message(
+            sender=sender,
+            recipient=recipient,
+            content="whatever",
+            date_sent=timezone_now(),
+            sending_client=sending_client,
+            last_edit_time=timezone_now(),
+            edit_history="[]",
+        )
+        message.set_topic_name("whatever")
+        message.save()
+
+        for i, emoji in enumerate(["a", "b", "c", "d", "e"]):
+            reaction = Reaction(
+                user_profile=sender, message=message, emoji_name=emoji, emoji_code=i
+            )
+            reaction.save()
```

Comment:
Similarly with these, we don't want to add reactions directly using the ORM, even in tests.  An OK option would be to call `do_add_reaction` to create the emoji reactions.  Better would be to make an actual API call, which would require using valid reaction names/codes; see `zerver/tests/test_reactions.py` for examples of doing this.

Reply:
This test seems like more of a unit test of the `Reaction.get_raw_db_rows` method.

Perhaps I can add an integration test to `test_reactions.py` as well.

Upon second thought, I might argue that a unit test of the ORM method might not even be worth testing since it is unlikely to catch regressions anyway. Regressions could happen anywhere on the stack all the way down to the API call. Maybe I'll kill this test entirely.